src/Microsoft.Health.Fhir.Shared.Api/Features/Resources/Bundle/BundleHandler.cs

```diff
@@ -207,8 +211,9 @@ private static void AddHeaderIfNeeded(string headerKey, string headerValue, Http
 
         private async Task ExecuteRequests(Hl7.Fhir.Model.Bundle responseBundle, Hl7.Fhir.Model.Bundle.HTTPVerb httpVerb)
         {
-            foreach (RouteContext request in _requests[httpVerb])
+            foreach (var tuple in _requests[httpVerb])
```

Comment:
nit: for readability we could use deconstruction here -- so something like `foreach ((RouteContext routeContext, int entryIndex) in _requests[httpVerb])`